src/soc/intel/common/block/fast_spi/mmap_boot.c

```diff
@@ -114,13 +114,24 @@ static void bios_mmap_init(void)
 	_Static_assert(CONFIG_EXT_BIOS_WIN_SIZE != 0, "Extended BIOS window size cannot be 0!");
 
 	/*
+	 * CONFIG(EXT_BIOS_FILL_UP) is not set
+	 * -----------------------------------
+	 *
 	 * Remaining portion of the BIOS region up to a maximum of CONFIG_EXT_BIOS_WIN_SIZE is
 	 * mapped at the top of the extended window if the BIOS region is greater than 16MiB.
 	 *
 	 * If the BIOS region is not greater than 16MiB, then the extended window is not
 	 * enabled.
+	 *
+	 * CONFIG(EXT_BIOS_FILL_UP) is set
+	 * -------------------------------
+	 *
+	 * Same as above, but part of the flash preceding the BIOS region gets mapped into the
+	 * extended window as well (as much of it as there is room left).
```

Comment:
When BIOS region is smaller than 16M, the rest of flash is still mapped into extended window. This either must be documented (with at least a note on diagram above the function that this is special case, as well as a comment here) or code changed to add `bios_start` to `bios_size` right after it is obtained, so it will be used for calculation of fixed window.

Reply:
New version: d1b1e450f383703746d1d28fdafb02b173fc2c0c